internal/storage/blob/store.go

```diff
@@ -305,6 +386,96 @@ func (s *Store) pollForUpdates(ctx context.Context) {
 	}
 }
 
+func (s *Store) createSymlink(destination, source string) error {
+	if _, err := os.Lstat(source); err == nil {
+		if err := os.Remove(source); err != nil {
+			return fmt.Errorf("failed to delete left-over symlink at %q: %w", source, err)
+		}
+	}
+
+	//nolint:mnd
+	if err := os.MkdirAll(filepath.Dir(source), 0o744); err != nil {
```

Comment:
We don't necessarily need to mirror the directory layout of the remote bucket and could just create the symlinks in a flat structure. That would simplify things for us but, the error messages would be confusing for the users to interpret 🤔

Reply:
Wouldn't that break when there are two files with the same name? (such as `dir/some.yaml` and `some.yaml`)

Reply:
I was thinking of using the full path and replacing the path separator with another character. So, you'd have `dir|some.yaml` and `some.yaml`. It's probably too much though.